src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IHandlerProvider.cs

```diff
@@ -16,5 +14,5 @@ public interface IHandlerProvider
 {
     ImmutableArray<RequestHandlerMetadata> GetRegisteredMethods();
 
-    IMethodHandler GetMethodHandler(string method, Type? requestType, Type? responseType);
+    ImmutableArray<Lazy<IMethodHandler, string?>> GetMethodHandlers(string method, Type? requestType, Type? responseType);
```

Comment:
As before, I think adding the optional language as an input here and keeping the same return value would make more sense

Reply:
Done